Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -1003,20 +1035,61 @@ protected int GetVoiceCommandIndex(string command)
 
         #endregion VoiceCommands
 
-        #region IMixedRealityTouchHandler
+        #region NearInteractionHandlers
+        /// <summary>
+        /// Is the eventData a grab action or near pointer?
+        /// </summary>
+        /// <param name="eventData"></param>
+        /// <returns></returns>
+        protected bool IsGrab(IMixedRealityPointer pointer, MixedRealityInputAction action)
+        {
+            if (pointer as IMixedRealityNearPointer != null)
```

Comment:
All this will return is whether the interaction is a near pointer, not if it is a grab action.

If you wanted to check for a grab action the way to do it would be to define a sister field to "Input Actions" and compare against that action. Something like this:

```
// Rename "InputActions" to "select action" since this is the MR action we want the interactable to respond to
MixedRealityAction selectAction;
// All events that match the grab actions will be considered grabs
MixedRealityAction grabAction;

bool isGrab(pointer)
{
    return pointer.action == grabAction;
}
```

However, it's possible that you don't actually want to distinguish between grab and select actions like this, since if I understand Interactable correctly, the way to distinguish this would just be to have two different interactables -- one for grab action and one for the select action. Happy to explain more my understanding.

Perhaps what you reall want here is to distinguish between near and far interactions, and to have corresponding "near interaction states" instead of "grip states"

Reply:
I would like Grab to reflect near grab or grab on controller, otherwise, we are a far interaction press (unless controller). Having two MixedRealityAction dropdowns on the Interactable could add confusion for what may be a minimal use-case. I was thinking about having another component handle the grab from a controller and routing that in. I'm worried about getting too descriptive of what a grab is and is not, because I could be too limiting. I would love to expose more control over how these states are defined or let people route them in.

Reply:
When you say "grab on controller", do you mean "the grip button on the controller is pressed when using far interaction"? Interestingly, the current design of the BaseControllerPointer is actually *only* letting the select button to go through the pointer system, and I think very few people know that. The fact that pointers only dispatch when select button is pressed is also causing #4492.

This method as written will actually return false if you try to interact with pointer via the grab button. This is because the method currently just returns true if the pointer is a near pointer. And until #4492 is fixed you won't get events when you press the grab button anyway. 

When I understood the original description from @cre8ivepark , I actually thought when he said "grab" he was meaning "near interaction", as in "hand / controller was near object when the 'select' button pressed". If dev configured interactable to respond to a button other than select, for example the menu button, then it would read "hand/controller was near object when the 'menu' button was pressed".

Reply:
I may be looking at this too deeply. The currently implimentation is looking at Grab by testing if a near interaction is in the pressed state. It only works if the NearInteractionGrabbable is present, so this is working as expected.

Do we really need to differentiate the grab on a controller from any other action or will a press event be enough? As @julenka mentioned, it would have the selected action from the Input Action drop-down. I don't like the idea of trying to detect mulitple input types if we have an input action selector to filter input. In this case we are detected near interactions touch and grab, not buttons on a controller, using the select action. This makes sense to me. Is it OK to say Grab for near Interaction Grab, and not be confused with Grip from a controller? We could say Physical Grab, like Physical Touch? I don't like adding words if they are not needed, just throwing that out there for consistency.

Thanks for all the great feedback, you are definitely asking all the right questions.

I'll remove all the controller grip stuff and assume that if someone wants grip they will select it from the input action drop-down and use the press state, like all the other input types. This clears up things for me at least.

Reply:
It is interesting that the Near Interaction Grab sends a Grip Press action though, but I filter this by checking for the NearPointer.